seastar/src/config_and_start_seastar.rs

```diff
@@ -0,0 +1,390 @@
+        AppTemplate {
+            app: new_app_template_from_options(opts.opts.pin_mut()),
+        }
+    }
+
+    /// Runs an app with a void callback (the output of which is always 0) and program arguments (argv).
+    ///
+    /// Currently, this function can only be called once in a single thread.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::AppTemplate;
+    ///
+    /// let fut = async move { Ok(()) };
+    ///
+    /// let mut app = AppTemplate::default();
+    /// let args = vec!["hello"];
+    ///
+    /// assert_eq!(app.run_void(&args[..], fut), 0);
+    /// ```
+    pub fn run_void<I, Arg>(
+        &mut self,
+        args: I,
+        fut: impl Future<Output = cxx_async::CxxAsyncResult<()>> + 'static,
+    ) -> i32
+    where
+        I: IntoIterator<Item = Arg>,
+        Arg: Into<OsString>,
+    {
+        let args = get_c_args(args);
+        let mut args: Vec<_> = args.iter().map(|s| s.as_ptr() as *mut c_char).collect();
+        unsafe {
+            run_void(
+                self.app.pin_mut(),
+                args.len() as i32,
+                args.as_mut_ptr(),
+                VoidFuture::fallible_local(fut),
+            )
+        }
+    }
+
+    /// Runs an app with an int (status code) callback and program arguments (argv).
+    ///
+    /// Currently, this function can only be called once in a single thread.
+    ///
+    /// # Examples
+    ///
+    /// ```rust
+    /// use seastar::AppTemplate;
+    ///
+    /// let fut = async move { Ok(42) };
+    ///
+    /// let mut app = AppTemplate::default();
+    /// let args = vec!["hello"];
+    ///
+    /// assert_eq!(app.run_int(&args[..], fut), 42);
+    /// ```
+    pub fn run_int<I, Arg>(
+        &mut self,
+        args: I,
+        fut: impl Future<Output = cxx_async::CxxAsyncResult<i32>> + 'static,
+    ) -> i32
+    where
+        I: IntoIterator<Item = Arg>,
+        Arg: Into<OsString>,
+    {
+        let args = get_c_args(args);
+        let mut args: Vec<_> = args.iter().map(|s| s.as_ptr() as *mut c_char).collect();
+        unsafe {
+            run_int(
+                self.app.pin_mut(),
+                args.len() as i32,
+                args.as_mut_ptr(),
+                IntFuture::fallible_local(fut),
+            )
+        }
+    }
+}
+
+impl Default for AppTemplate {
+    fn default() -> Self {
+        AppTemplate::new_from_options(Options::default())
+    }
+}
+
+fn get_c_args<I, Arg>(iter: I) -> Vec<CString>
+where
+    I: IntoIterator<Item = Arg>,
+    Arg: Into<OsString>,
+{
+    use std::os::unix::ffi::OsStringExt;
+    iter.into_iter()
+        .map(|os| {
+            let v = os.into().into_vec();
+            CString::new(v).unwrap_or_else(|err| {
+                let nul_position = err.nul_position();
+                let mut v = err.into_vec();
+                v.truncate(nul_position);
+                CString::new(v).unwrap()
+            })
+        })
+        .collect()
+}
```

Comment:
I read on SO that `char** argv[]` has to be null terminated (have `argv[argc] = NULL`)
https://stackoverflow.com/a/35335248

With Vec<CString> this wouldn't be the case.

Reply:
I didn't know about that, thanks. This makes `argc` a bit redundant, but if the null pointer in the end is required - so be it.